test-utils/src/main/groovy/io/micronaut/starter/test/github/WorkflowSpec.groovy

```diff
@@ -0,0 +1,171 @@
+package io.micronaut.starter.test.github
+
+import groovy.util.logging.Slf4j
+import io.micronaut.core.util.StringUtils
+import io.micronaut.starter.application.Project
+import io.micronaut.starter.feature.github.workflows.Secret
+import io.micronaut.starter.test.CommandSpec
+import io.micronaut.starter.test.github.client.GitHubApiClient
+import io.micronaut.starter.test.github.client.GitHubRepository
+import io.micronaut.starter.test.github.client.GitHubSecret
+import io.micronaut.starter.test.github.client.GitHubSecretsPublicKey
+import io.micronaut.starter.test.github.client.GitHubUser
+import io.micronaut.starter.test.github.client.GitHubWorkflowRun
+import io.micronaut.starter.test.github.client.GitHubWorkflowRuns
+import org.apache.tuweni.crypto.sodium.Box
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.api.errors.GitAPIException
+import org.eclipse.jgit.transport.PushResult
+import org.eclipse.jgit.transport.RemoteRefUpdate
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider
+import spock.lang.Shared
+import spock.util.concurrent.PollingConditions
+
+import java.nio.file.Path
+import java.util.stream.Collectors
+import java.util.stream.StreamSupport
+
+@Slf4j
+abstract class WorkflowSpec extends CommandSpec{
+
+    private static final String TOKEN_PREFIX = "token ";
+    public static final String GH_TOKEN = "GH_TOKEN"
+
+    @Shared
+    private String apiToken;
+
+    @Shared
+    private String token;
+
+    void setupSpec() {
+        token = System.getenv(GH_TOKEN)
+        apiToken = TOKEN_PREFIX + token;
+    }
+
+    protected GitHubApiClient getGitHubApiClient(){
+        return beanContext.getBean(GitHubApiClient)
+    }
+
+    protected void pushToGitHub(Project project, List<Secret> secrets){
```

Comment:
This basically duplicates in a lot the starter-api code of push-to-github. Would like to hear your opinion what is the preferred way:
A) dependency of `starter-api` in `test-utils`
B) moving the shared code into `starter-core`
C) create new module `starter-github`

cc: @graemerocher @alvarosanchez ?

Reply:
My opinion would be B)

Reply:
ok will move the code today .. please check the rest of the code